### PR TITLE
DeviceTelemetry event grid update

### PR DESCRIPTION
**Pull request by kapil-ms**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5736 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5736/commits/862ca433404fc4cfbd5afe8355feae4017fdec1c

**Comment by kapil-ms** [comment]
Couple of open questions-
1.	Can the leaf objects in a swagger document be of object type or have to be primitive type ?
2.	The body can be an object or a string. Does swagger identify string as an object as well ?

**Comment by kapil-ms** [comment]
@Kishp01 @kalyanaj

File: specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json [hunk 1]
```diff
@@ -44,6 +43,15 @@
       ],
       "properties": {}
     },
+    "IotHubDeviceTelemetryEventData": {
```
Comment (sarangan12) [inline specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json]: Can you tell me where this is used or referenced? I do not find anywhere this is used.

**Review by sarangan12 — CHANGES_REQUESTED** [review]

File: specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json (hunk 1 above)
Comment (kapil-ms) [reply specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json]: That could be said for other definitions as well like IotHubDeviceCreatedEventData, IotHubDeviceCreatedEventData, etc. Is there something I'm missing ?

Comment (sarangan12) [reply specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json]: Sure. The same question can be asked for the definition IotHubDeviceCreatedEventData. That is the reason I am asking. WHat is the point of these definitions? How exactly a SDK customer will use this definition (which will be a class in the generated SDK?)

Comment (kapil-ms) [reply specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json]: @Kishp01 @kalyanaj

Comment (kalyanaj) [reply specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json]: These definitions become part of the generated EventGrid data plane SDKs. For example, using Event Grid, a customer can create an "event subscription" subscribing to an IotDeviceCreated event and providing a destination such as a HTTPS endpoint (e.g. an Azure function). EventGrid will deliver such events to the customer's endpoint. The customer's endpoint can then consume the received event using this EventGrid data plane SDK. For a sample (that uses StorageBlobCreatedEventData), please see https://github.com/Azure-Samples/event-grid-dotnet-publish-consume-events/blob/master/EventGridConsumer/EventGridConsumer/Function1.cs.

**Review by sarangan12 — APPROVED** [review]

**Review by kalyanaj — APPROVED** [review]
Looks good to me.

**Pull request merged**